src/components/InputStepCounter/InputStepCounter.tsx

```diff
@@ -140,6 +144,11 @@ const StepCounter = ({
     }
   }, [localValue, useLocalValue, value])
 
+  const baseCurrencyPriceInUSDC = useUSDCPrice(currencyB ?? undefined)
```

Comment:
sorry to be pedantic but this is a bit hard to follow. Could you move it to a custom hook  at the top of this file? Something like

```
function useBaseCurrencyAmount(baseCurrency: Currency | undefined, quoteAmount: number) {
  const currencyUSDCPrice = useUSDCPrice(baseCurrency ?? undefined)
  if (!currencyUSDCPrice) return undefined
  return Number(currencyUSDCPrice.toSignificant(6)) * quoteAmount
}
```